HASH/funcionesLael.c

```diff
@@ -0,0 +1,239 @@
+/**
+*================================================================================
+*@file   mapas2.c
+*
+*@brief  Este modulo contiene parte de las funciones correspondientes al 
+*        funcionamiento de mapas en C, aqui podran econtrar la documentacion 
+*        individual asi como el codigo de las siguientes funciones:
+*
+*        containsValue, entrySet, hashCode, isEmpty putAll.
+*
+*@author 
+*================================================================================
+*/
+
+#include "bibliotecas.h"
+
+//FUNCIONES GENERALES DEL PROGRAMA
+extern void limpia(void);
+extern int ascii(char llave[]);
+
+// FUNCIONES DESARROLLADAS POR LAEL
+int containsValue(MAPA *mapa[],char contenido[]);  
+void entrySet(MAPA *mapa[], MAPA **raiz);  
+int hashCode(MAPA *mapa[]);  
+int isEmpty(MAPA *mapa[]);  
+void putAll(MAPA *mapa1[], MAPA *mapa2[]);  
+
+//================================================================================================|||||||||||||||||||||||||||
+//FUNCIONES DESARROLLADAS POR LAEL
+
+/**
+ *===============================================================================
+ *@brief     Esta funcion almacena en un archivo binario espeficificado en el main
+ *           por el usuario todos los registros de los mapas o tambien los 
+ *           actualiza.
+ *
+ *@entradas  MAPA *m[]: Mapa donde el nuevo elemento sera ingresado.   
+ *           char contenido[]: Contenido del elemento nuevo.
+ *
+ *@salidas   0: Valor verdadero, si encontro el cotenido a buscar
+ *           1: Valor falso, si no encontro el contenido a buscar
+ *
+ *@variables *nodo: Apuntador a estructura del mapa.  
+ *           *aux:  Apuntador de soporte para "*nodo"
+ *           i: Contador de FOR.
+ *
+ *@autor     Leonardo Lael Villar Benitez
+ *===============================================================================
+ */
+int containsValue(MAPA *mapa[],char contenido[])
+{
+	MAPA *nodo, *aux;
+	int i;
+	for(i=0; i<T; i++)  //RECORRIDO DEL MAPA
+	{
+		if(mapa[i] != NULL)  //SI EXITEN ENTRADAS
+		{
+			nodo = mapa[i];
+			aux = nodo;
+			if(strcmp(contenido,nodo->CONTENIDO) == 0) //COMPARACION CON EL CONTENIDO A BUSCAR
+			{
+				return 0;
+			}
+			if(aux->sig != NULL)  //SI TIENE NODOS ENCADENADOS...
+			{
+				while(aux->sig != NULL)
+				{
+					aux = aux->sig;
+					if(strcmp(contenido,aux->CONTENIDO) == 0)  //COMPARACION CON EL CONTENIDO A BUSCAR
+					{
+						return 0;
+					}
+				}
+			}
+		}
+	}
+	return 1;  //SI NO ENCONTRO EL CONTENIDO A BUSCAR
+} 
```

Comment:
Yo se que funciona containsValue, pero generalmente quieres evitar mas de 3 if's anidados.
a la mejor si inviertes un poco tu logica, mejora el codigo.

como detalle te dejo este link, esto lo quieres evitar a toda costa
http://thedailywtf.com/articles/finding-the-file